### ISSUE TITLE
Renaming title doesn't update links

### ISSUE THREAD

**Issue by 3Rafal** [org-roam/org-roam#1425]
### Description
When renaming title, we call hook
```
(defcustom org-roam-title-change-hook '(org-roam--update-file-name-on-title-change
                                        org-roam--update-links-on-title-change)
```
that should change filename and update links.
The filename is changed correctly, but links doesn't seem to update.
This was supposed to be fixed on #1073 , but I guess it was still problematic to @shlevy (as reported in comment under this PR)
Also, it seems that this bug happened to @zaeph in EmacsConf presentation.

#### Steps to Reproduce
1. Create a file
2. Create a link to this file
3. Rename file title

#### Expected Results
Both filename and link should be updated

#### Actual Results
Only filename is renamed. Link refers to old name

### Environment

- Emacs: GNU Emacs 27.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.20, cairo version 1.16.0)
 of 2020-09-19
- Framework: N/A
- Org: Org mode version 9.3 (release_9.3 @ /usr/share/emacs/27.1/lisp/org/)
- Org-roam: 1.2.3

**Comment by dndln86stone** [comment]
I have similar problem with a little difference.

1. create file `aaa` with `org-roam-find-file`.
2. make link `aaa` in another file `bbb`.
3. change title of `aaa` to `aaa2`
4. filename `2021...-aaa.org` is changed to `2021...-aaa2.org` and,
link in file `bbb` is changed to `aaa2` correctly.
5. **close emacs and restart.**
6. then, when I change title of `aaa2`, the filename and link in file `bbb` **are not changed.**

**Comment by dndln86stone** [comment]
I think my problem is different with @3Rafal .
My problem can be fixed by opening org-roam file with `org-roam-find-file`.

Starting emacs with org-roam file `aaa2` from terminal like `emacs aaa2.org`, in my case, `org-roam--handle-title-change` is not inserted to `after-save-hook`.

So I open another org-roam file, and close `aaa2` and open `aaa2` using `org-roam-find-file`.
Then `after-save-hook` contains `org-roam--handle-title-change`, and title rename works properly for both filename and link.

> I have similar problem with a little difference.
> 
> 1. create file `aaa` with `org-roam-find-file`.
> 2. make link `aaa` in another file `bbb`.
> 3. change title of `aaa` to `aaa2`
> 4. filename `2021...-aaa.org` is changed to `2021...-aaa2.org` and,
>    link(both of link and description) in file `bbb` is changed to `aaa2` correctly.
> 5. **close emacs and restart.**
> 6. then, when I change title of `aaa2`, the filename and link(both of link and description) in file `bbb` **are not changed.**

**Comment by 3Rafal** [comment]
Ok, so I guess I have figured out why my renames didn't work.
As far as I know, the rename mechanism uses cache, which is by default updated after 2 seconds of user being idle.
In my scenario, I sometimes did the renames without a pause, so the cache was not updated.
I switched to `(setq org-roam-db-update-method 'immediate)` and it works. 

I wonder if there is a way to state this in docs, so that it will be easier for other to figure out. Showing a elation between cache and consistency.
